actor/builtin/miner/miner.go

```diff
@@ -542,6 +554,51 @@ func (ma *Actor) CommitSector(ctx exec.VMContext, sectorID uint64, commD, commR,
 	return 0, nil
 }
 
+// VerifyPieceInclusion verifies that proof proves that the data represented by commP is included in the sector.
+// This method returns nothing if the verification succeeds and returns a revert error if verification fails.
+func (ma *Actor) VerifyPieceInclusion(ctx exec.VMContext, commP []byte, sectorID uint64, proof []byte) (uint8, error) {
```

Comment:
COMMENT: I was surprised to see that the PIP was an input to this method and not part of the `commitSector` message. I was surprised to see this because it means that a (CommP, PIP) tuple will need to go on chain each time `VerifyPieceInclusion` is called. The `commitSector` will appear on chain only once per SectorID. I **thought** that `verifyPieceInclusion` would be included in the chain each time funds would be released. I was wrong.

What I learned is that a `verifyPieceInclusion` message will be posted on chain only when the storage client challenges the miner to demonstrate that their piece has been included in a sector for which they've generated a (recent) PoSt. Challenging will cost storage client, and we don't expect them to do this often. The best case scenario will see `verifyPieceInclusion` included in the chain 0 times per CommP/SectorID pair. Because of this, we do not think that it makes much sense to include the CommP/PIP mappings in the `commitSector` message (which, of course, will always go on-chain).